pkg/ccl/streamingccl/streamingest/testdata/span_configs

```diff
@@ -0,0 +1,113 @@
+# This test asserts that the span configs associated with the replicating app tenant also get
+# replicated
+
+create-replication-clusters
+----
+
+exec-sql as=source-system
+ALTER TENANT ALL SET CLUSTER SETTING spanconfig.reconciliation_job.checkpoint_interval = '10ms'
+----
+
+
+exec-sql as=source-tenant
+CREATE TABLE d.x (id INT PRIMARY KEY, n INT);
+----
+
+exec-sql as=source-tenant
+CREATE TABLE d.x2 (id INT PRIMARY KEY, n INT);
+----
+
+
+# Get the mininum and maximum user table id
+let $min_table_id  as=source-tenant
+SELECT 'd.x'::regclass::oid::int::string
+----
+
+let $max_table_id  as=source-tenant
+SELECT 'd.x2'::regclass::oid::int::string
+----
+
+
+start-replication-stream
+----
+
+# TODO(adityamaru): This is a hack until we have the time we started replicating
+# from persisted in the replication job details.
+let $start as=source-system
+SELECT clock_timestamp()::timestamp::string
+----
+
+wait-until-replicated-time ts=$start
+----
+
+list-ttls as=source-system min_table_id=$min_table_id max_table_id=$max_table_id
+----
+14400
+14400
+
+exec-sql as=source-tenant
+ALTER TABLE d.x CONFIGURE ZONE USING gc.ttlseconds=1234;
+----
+
+list-ttls as=source-system min_table_id=$min_table_id max_table_id=$max_table_id
+----
+1234
+14400
+
+
+# FIGURE OUT HOW TO COMBINE THE SPANCONFIGS, CAUSING A DELETE
```

Comment:
:D Let's make this a TODO.

Can we cause a delete by dropping the table gc.ttl on the table and then dropping the table?

Reply:
Added a Todo. Chatting with Arul about this. It's harder than I initially thought.